PiT 24.4: InvalidPathException in DevModeInitializer.getPhysicalFileOfJBossVfsJar when running CDI project in windows,

### Description of the bug

When upgrading `skeleton-starter-flow-cdi` to 24.4.0 it fails in windows, error is:

```
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 3: vfs:/D:/a/platform/platform/tmp/skeleton-starter-flow-cdi/content/project-base-flow-cdi-1.0-SNAPSHOT.war/WEB-INF/lib/copilot-24.4.0.alpha132603338928158166396.jar
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:232)
	at java.base/java.nio.file.TempFileHelper.generatePath(TempFileHelper.java:57)
	at java.base/java.nio.file.TempFileHelper.create(TempFileHelper.java:125)
	at java.base/java.nio.file.TempFileHelper.createTempFile(TempFileHelper.java:159)
	at java.base/java.nio.file.Files.createTempFile(Files.java:923)
	at deployment.project-base-flow-cdi-1.0-SNAPSHOT.war//com.vaadin.base.devserver.startup.DevModeInitializer.getPhysicalFileOfJBossVfsJar(DevModeInitializer.java:474)
	at deployment.project-base-flow-cdi-1.0-SNAPSHOT.war//com.vaadin.base.devserver.startup.DevModeInitializer.getFrontendLocationsFromClassloader(DevModeInitializer.java:395)
	at deployment.project-base-flow-cdi-1.0-SNAPSHOT.war//com.vaadin.base.devserver.startup.DevModeInitializer.getFrontendLocationsFromClassloader(DevModeInitializer.java:344)
	at deployment.project-base-flow-cdi-1.0-SNAPSHOT.war//com.vaadin.base.devserver.startup.DevModeInitializer.initDevModeHandler(DevModeInitializer.java:240)
	at deployment.project-base-flow-cdi-1.0-SNAPSHOT.war//com.vaadin.base.devserver.DevModeHandlerManagerImpl.initDevModeHandler(DevModeHandlerManagerImpl.java:96)
	at deployment.project-base-flow-cdi-1.0-SNAPSHOT.war//com.vaadin.base.devserver.startup.DevModeStartupListener.initialize(DevModeStartupListener.java:78)
	at deployment.project-base-flow-cdi-1.0-SNAPSHOT.war//com.vaadin.flow.server.startup.VaadinServletContextStartupInitializer.process(VaadinServletContextStartupInitializer.java:42)
	at deployment.project-base-flow-cdi-1.0-SNAPSHOT.war//com.vaadin.flow.server.startup.ClassLoaderAwareServletContainerInitializer.lambda$onStartup$2(ClassLoaderAwareServletContainerInitializer.java:81)
```

### Expected behavior

No failure like in mac or linux

### Minimal reproducible example

In a windows machine unpack attached project, and run `./mvnw.cmd wildfly:run`
[skeleton-starter-flow-cdi.zip](https://github.com/vaadin/flow/files/14762061/skeleton-starter-flow-cdi.zip)

The project has been generated by running:

```
git clone https://github.com/vaadin/skeleton-starter-flow-cdi.git
cd skeleton-starter-flow-cdi
perl -0777 -pi -e 's|(<vaadin.version>)([^\s]+)(</vaadin.version>)|${1}24.4.0.alpha17${3}|g' pom.xml
perl -pi -e 's|(\s*)(</properties>)|$1$2\n$1<repositories><repository><id>v</id><url>https://maven.vaadin.com/vaadin-prereleases</url></repository></repositories>|' pom.xml
perl -pi -e 's|(\s*)(</properties>)|$1$2\n$1<pluginRepositories><pluginRepository><id>v</id><url>https://maven.vaadin.com/vaadin-prereleases</url></pluginRepository></pluginRepositories>|' pom.xml
perl -0777 -pi -e 's!(\s+)(<artifactId>(vaadin|vaadin-quarkus-extension)</artifactId>)!${1}${2}${1}<exclusions>${1}    <exclusion>${1}        <groupId>com.vaadin</groupId>${1}        <artifactId>hilla-dev</artifactId>${1}    </exclusion>${1}</exclusions>!' pom.xml
./mvnw.cmd -ntp -B wildfly:run  
```

### Versions

- Vaadin / Flow version: 24.4.0.alpha17
- OS version: Windows

--- a/vaadin-dev-server/src/main/java/com/vaadin/base/devserver/startup/DevModeInitializer.java
+++ b/vaadin-dev-server/src/main/java/com/vaadin/base/devserver/startup/DevModeInitializer.java
@@ -469,7 +469,7 @@ public class DevModeInitializer implements Serializable {
             String vfsJarPath = url.toString();
             String fileNamePrefix = vfsJarPath.substring(
                     vfsJarPath.lastIndexOf(
-                            FrontendUtils.isWindows() ? '\\' : '/') + 1,
+                            vfsJarPath.contains("\\") ? '\\' : '/') + 1,
                     vfsJarPath.lastIndexOf(".jar"));
             Path tempJar = Files.createTempFile(fileNamePrefix, ".jar");
 

--- a/vaadin-dev-server/src/test/java/com/vaadin/base/devserver/OpenInCurrentIdeTest.java
+++ b/vaadin-dev-server/src/test/java/com/vaadin/base/devserver/OpenInCurrentIdeTest.java
@@ -9,8 +9,11 @@ import java.util.List;
 import java.util.Optional;
 
 import org.junit.Assert;
+import org.junit.Assume;
 import org.junit.Test;
 import org.mockito.Mockito;
+
+import com.vaadin.open.OSUtils;
 
 public class OpenInCurrentIdeTest {
 
@@ -46,7 +49,8 @@ public class OpenInCurrentIdeTest {
         // The binary on Mac is /.../IntelliJ IDEA.app/Contents/MacOS/idea
         Assert.assertEquals(
                 new File(baseDirectory, "MacOS/idea").getAbsolutePath(),
-                OpenInCurrentIde.getBinary(ideCommand.get()));
+                new File(OpenInCurrentIde.getBinary(ideCommand.get()))
+                        .getAbsolutePath());
 
     }
 
@@ -605,6 +609,8 @@ public class OpenInCurrentIdeTest {
 
     @Test
     public void runThrowsExceptionOnFailure() throws InterruptedException {
+        Assume.assumeFalse(OSUtils.isWindows());
+
         try {
             OpenInCurrentIde.run("/bin/sh", "-c",
                     "echo 'output1'; echo 'output2'; exit 123");

